### ISSUE TITLE
How to convert these PyTorch functions to TensorFlow.js (TFJS)?

### ISSUE THREAD

**Issue by lancejpollard** [tensorflow/tfjs#7308]
I am looking to port something from PyTorch to Tensorflow.js (TFJS), and could use some help in making sure I get the functions mapped correctly from one framework to the other. I have already started, for example both frameworks have the same `torch.where` and `tf.where` function, and `torchTensor.clamp` is `tf.clipByValue`. But some of the others are harder to find and I'm not sure the exact mapping.

#### Questionable

- `torchTensor.isclose -> ?`
- `torch.all -> ?`
- `torchTensor.clamp_max -> ?`
- `torchTensor.gt -> ?` https://pytorch.org/docs/stable/generated/torch.gt.html#torch.gt
- `torchTensor.lt -> ?`
- `torchTensor.any -> ?` https://stackoverflow.com/questions/53401053/is-there-all-or-any-equivalent-in-python-tensorflow
- `torch.unsqueeze -> ?` https://pytorch.org/docs/stable/generated/torch.unsqueeze.html#torch.unsqueeze
- `torch.broadcastTensors -> ?`
- `torchTensor.dim -> ?` https://stackoverflow.com/questions/36966316/how-to-get-the-dimensions-of-a-tensor-in-tensorflow-at-graph-construction-time
- `torch.narrow -> ?` https://pytorch.org/docs/stable/generated/torch.narrow.html#torch.narrow
- `torch.masked_fill -> ?` ([this](https://stackoverflow.com/questions/47447272/does-tensorflow-have-the-function-similar-to-pytorchs-masked-fill))

      def mask_fill_inf(matrix, mask):
          negmask = 1 - mask
          num = 3.4 * math.pow(10, 38)
          return (matrix * mask) + (-((negmask * num + num) - num))

#### Figured Out

- `torch.numel -> tf.shape.num_elements` https://www.tensorflow.org/api_docs/python/tf/TensorShape#num_elements
- `torchTensor.norm -> tf.norm`
- `torch.cat -> tf.concat`
- `torch.prod -> tf.math.reduce_prod`
- `torch.squeeze -> tf.squeeze`
- `torch.zeros -> tf.zeros` (but doesn't have the `device` option)
- `torchTensor.reciprocal -> tf.math.reciprocal`
- `torchTensor.size -> tf.size`

Basically I am porting [this file](https://github.com/geoopt/geoopt/blob/27a1e6fd825e2de923b8493bb8baf514f3f089ea/geoopt/manifolds/stereographic/math.py) form PyTorch to TensorFlow, so these are the functions it uses.

It appears that `all -> tf.math.reduce_all` in _Python_, but there is no `reduceAll` in TFJS, what am I missing?

Thank you very much for your time and any and all help.

**Comment by pyu10055** [comment]
@lancejpollard I took a quick look at the missing ops you mentioned, there are an equivalent tfjs op/ops, you can take a close look at [tfjs api doc](https://js.tensorflow.org/api/latest), and play these ops on the page.

torchTensor.isclose -> ? you can use [tf.equal()](https://js.tensorflow.org/api/latest/#equal)

torch.all -> ?  tfjs has [tf.all](https://js.tensorflow.org/api/latest/#all)

torchTensor.clamp_max -> ? you can use [clipByValue](https://js.tensorflow.org/api/latest/#clipByValue) with a Number. MIN_VALUE

torchTensor.gt -> ? https://pytorch.org/docs/stable/generated/torch.gt.html#torch.gt
[tf.greater](https://js.tensorflow.org/api/latest/#greater)

torchTensor.lt -> ? [tf.less](https://js.tensorflow.org/api/latest/#less)

torchTensor.any -> ? https://stackoverflow.com/questions/53401053/is-there-all-or-any-equivalent-in-python-tensorflow
[tf.any](https://js.tensorflow.org/api/latest/#any)

torch.unsqueeze -> ? https://pytorch.org/docs/stable/generated/torch.unsqueeze.html#torch.unsqueeze
[tf.expandDim](https://js.tensorflow.org/api/latest/#expandDims)
torch.broadcastTensors -> ?
[tf.broadcastTo](https://js.tensorflow.org/api/latest/#broadcastTo)
torchTensor.dim -> ? https://stackoverflow.com/questions/36966316/how-to-get-the-dimensions-of-a-tensor-in-tensorflow-at-graph-construction-time
tensor.rank
torch.narrow -> ? https://pytorch.org/docs/stable/generated/torch.narrow.html#torch.narrow
[tf.stridedSlice](https://js.tensorflow.org/api/latest/#stridedSlice)
torch.masked_fill -> ? ([this](https://stackoverflow.com/questions/47447272/does-tensorflow-have-the-function-similar-to-pytorchs-masked-fill))
mask_fill seems to be similar to tf.where
```
tf.where(mask, input, value)
```

**Comment by gaikwadrahul8** [comment]
Hi, @lancejpollard 

Apologize for the delayed response and could you please refer above comment from @pyu10055, Please let us know is it resolving your issue ? If yes please feel free to close this issue. 

If issue still persists please let us know ? Thank you!
